gobblin-salesforce/src/main/java/gobblin/salesforce/SalesforceSource.java

```diff
@@ -217,8 +222,10 @@ private Histogram getHistogram(String entity, String watermarkColumn, SourceStat
         List<Command> commands = RestApiConnector.constructGetCommand(connector.getFullUri(soqlQuery));
         CommandOutput<?, ?> response = connector.getResponse(commands);
         histogram.add(parseHistogram(response));
+      } catch (DataRecordException e) {
+        throw e;
       } catch (Exception e) {
```

Comment:
How about specifying the following list of specific exceptions and throwing either a runtime exception or DataRecordException?
      } catch (RestApiClientException | RestApiProcessingException | DataRecordException e) {
        throw new RuntimeException("Fail to get data of year " + year + " from salesforce", e);
      }